python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,218 @@
+        length = self.read_int(stream)
+
+        if length < 0:
+            print("command length is less than zero", file=sys.stderr)
+            return None
+        s = stream.read(length)
+
+        if len(s) < length:
+            raise EOFError
+        unenc = base64.urlsafe_b64decode(s)
+        return self.loads(unenc)
+
+    @staticmethod
+    def dumps(obj):
+        protocol = 2
+        return cPickle.dumps(obj, protocol)
+
+    @staticmethod
+    def loads(obj):
+        return cPickle.loads(obj)
+
+    @staticmethod
+    def pickle_and_write(self, obj, stream):
+        serialized = self.dumps( obj)
+
+        if serialized is None:
+            raise ValueError("Serialized value should not be None.")
+        elif len(serialized) > (1 << 31):
+            raise ValueError("Cannot serialize object larger than 2G.")
+
+        self.write_int(len(serialized), stream)
+        stream.write(serialized)
+
+
+def main(in_file, out_file):
+    pickle_serializer = PickleSerializer()
+    try:
+        func = pickle_serializer.read_command(in_file)
+        tuple_size = pickle_serializer.read_int(in_file)
+        output_type = pickle_serializer.read_int(in_file)
+        is_flatmap = pickle_serializer.read_int(in_file)
+
+        if tuple_size < 1:
+            raise ValueError("size of tuple should not be less than 1 ")
+
+        while True:
+            num_tuples = pickle_serializer.read_int(in_file)
+            if num_tuples == SpecialLengths.END_OF_STREAM:
+                break
+            tuple_list = []
+            for j in range(num_tuples):
+                tuple_list.append(pickle_serializer.read_tuple(in_file, tuple_size))
+
+            retval = func(tuple_list)
+            if is_flatmap > 0 :
+                count = len(retval)
+                pickle_serializer.write_int(count, out_file)
+                for i in range(count):
+                    pickle_serializer.write_with_length(
+                        retval[i], out_file, output_type)
+            else:
+                pickle_serializer.write_with_length(retval, out_file, output_type)
+
+            out_file.flush()
+
+    except Exception:
+        try:
+            pickle_serializer.write_int(SpecialLengths.PYTHON_EXCEPTION_THROWN, out_file)
+            pickle_serializer.write_with_length(traceback.format_exc().encode(
+                "utf-8"), out_file, 5)
+            print(traceback.format_exc(), file=sys.stderr)
+        except Exception:
+            print("python process failed with exception: ", file=sys.stderr)
+            print(traceback.format_exc(), file=sys.stderr)
+
+
+if __name__ == '__main__':
+    # Read a local port to connect to from stdin
+    java_port = int(sys.stdin.readline())
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    sock.connect(("127.0.0.1", java_port))
+    infile = os.fdopen(os.dup(sock.fileno()), "rb", 65536)
+    outfile = os.fdopen(os.dup(sock.fileno()), "wb", 65536)
+    exit_code = 0
```

Comment:
Where is `exit_code` being returned from the process? Did you mean to call `sys.exit(exit_code)` somewhere?